packages/react-components/react-field/src/components/Field/useFieldStyles.ts

```diff
@@ -19,16 +19,23 @@ const useRootStyles = makeStyles({
   base: {
     display: 'grid',
     gridAutoFlow: 'row',
+    gridTemplateColumns: 'auto 1fr',
     justifyItems: 'start',
   },
 
   horizontal: {
     gridTemplateRows: 'auto auto auto auto',
-    gridTemplateColumns: '1fr 2fr',
+    gridTemplateColumns: '1fr auto 2fr',
+  },
+
+  fullWidth: {
+    gridColumnStart: '1',
+    gridColumnEnd: '-1',
   },
 
   secondColumn: {
     gridColumnStart: '2',
+    gridColumnEnd: '-1',
   },
```

Comment:
This layout is getting complex enough that it might warrant using gridTemplateAreas instead of the fullWidth and secondColumn classes, and adding `gridArea` to each of the slots.

WDYT? This might just be trading one complexity for another, but it is easier to understand IMO.

```ts
vertical: {
  gridTemplateColumns: 'auto 1fr',
  gridTemplateAreas: `
    "label label"
    "control control"
    "validationIcon validationMessage"
    "hint hint"
  `,
},

horizontal: {
  gridTemplateColumns: '1fr auto 2fr',
  gridTemplateAreas: `
    "label control control"
    "label validationIcon validationMessage"
    "label hint hint"
    "label . ."
  `,
}
```

If you do that, you can take out `gridAutoFlow: 'row',` and `gridTemplateRows: 'auto auto auto auto',` since they're no longer needed.

Reply:
I like it! I think it ends up being marginally more CSS/generated classnames since then the child nodes can't share the column-start/column-end values, but I agree it's still worth it b/c it is much easier to read :D.